dev_config/python/.pre-commit-config.yaml

```diff
@@ -37,7 +37,7 @@ repos:
     hooks:
       - id: mypy
         additional_dependencies:
-          [types-requests, types-setuptools, types-pyyaml, types-toml]
+          [types-requests, types-setuptools, types-pyyaml, types-toml, pydantic>=2]
```

Comment:
Note:
Added dependency for `mypy` so people need to use `make install-precommit-hooks` to make this change take effect.